Bottom: f45f90764cf4f64c582c69651723d13dd2381b06
Top:    9bdf763425a7579e4b5acf9cc4e2b9d564c8dd83
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-01-06 17:07:51 +0000

polygon.c: For some reason, the range computation in frac_circle appears to be wrong


---

diff --git a/src/polygon.c b/src/polygon.c
index ac9f638..0177070 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -405,8 +405,8 @@ frac_circle (PLINE * c, Coord X, Coord Y, Vector v, int fraction)
   e1 = (v[0] - X) * POLY_CIRC_RADIUS_ADJ;
   e2 = (v[1] - Y) * POLY_CIRC_RADIUS_ADJ;
 
-  /* NB: the caller adds the last vertex, hence the -1 */
-  range = POLY_CIRC_SEGS / fraction - 1;
+  /* XXX */ /* NB: the caller adds the last vertex, hence the -1 */
+  range = POLY_CIRC_SEGS / fraction;
   for (i = 0; i < range; i++)
     {
       /* rotate the vector */
